src/compressor/Compressor.h

```diff
@@ -44,7 +43,8 @@ class Compressor {
     COMP_ALG_LAST   //the last value for range checks
   };
 
-  inline static const std::map<const std::string, const CompressionAlgorithm> compression_algorithms {
+  using pair_type = std::pair<const char*, CompressionAlgorithm>;
```

Comment:
Also from the Realms of Pedanticiscm dept., I personally would like a single blank line between this type definition and the data in the next line. YMMV. :-)